# GDP and Internet Use 

Imports

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

5. Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first 6 rows.

In [2]:
gdp_df = pd.read_csv("data/gdp_percapita.csv")
print(gdp_df.head(6))

  Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2018  2190.240321              NaN
1     Afghanistan  2017  2202.570851              NaN
2     Afghanistan  2016  2200.764487              NaN
3     Afghanistan  2015  2212.750899              NaN
4     Afghanistan  2014  2249.253738              NaN
5     Afghanistan  2013  2264.317575              NaN


6. Repeat for the internet use dataset. Call this DataFrame internet_df. Take a look at the first six rows.

In [3]:
internet_df = pd.read_csv("data/internet_use.csv")
print(internet_df.head(6))

  Country or Area  Year             Value Value Footnotes
0     Afghanistan  2014              6.39             NaN
1     Afghanistan  2013               5.9             NaN
2     Afghanistan  2012  5.45454545454545             NaN
3     Afghanistan  2011                 5             NaN
4     Afghanistan  2010                 4             NaN
5     Afghanistan  2009              3.55             NaN


7. Look at the shape of each DataFrame - how many rows? How many columns?

In [4]:
print("GDP:", gdp_df.shape)
print("Internet:", internet_df.shape)

GDP: (6608, 4)
Internet: (4676, 4)


ANS: GDP: 6608 rows and 4 columns, Internet: 4676 rows and 4 columns

8. Take a look at the datatypes for the columns in each DataFrame.

In [5]:
print(gdp_df.dtypes)

Country or Area     object
Year                object
Value              float64
Value Footnotes    float64
dtype: object


ANS: Country or Area: object, Year: object, Value: float64, Value Footnotes: float64

9. Take a look at the last 10 rows of each DataFrame in turn.

In [6]:
print(gdp_df.tail(10))
print(internet_df.tail(10))

     Country or Area                                      Year        Value  \
6598        Zimbabwe                                      1997  3580.048793   
6599        Zimbabwe                                      1996  3520.430146   
6600        Zimbabwe                                      1995  3226.413930   
6601        Zimbabwe                                      1994  3263.934978   
6602        Zimbabwe                                      1993  3033.504852   
6603        Zimbabwe                                      1992  3054.889178   
6604        Zimbabwe                                      1991  3426.598094   
6605        Zimbabwe                                      1990  3324.348171   
6606   footnoteSeqID                                  Footnote          NaN   
6607               2  Excludes South Sudan after July 9, 2011.          NaN   

      Value Footnotes  
6598              NaN  
6599              NaN  
6600              NaN  
6601              NaN  
6602      

10. Drop the value footnotes column from both DataFrames. Check that this worked as expected.

In [7]:
gdp_df.drop('Value Footnotes', inplace = True, axis = 1)
print(gdp_df.head(5))
internet_df.drop('Value Footnotes', inplace = True, axis = 1)
print(internet_df.head(5))

  Country or Area  Year        Value
0     Afghanistan  2018  2190.240321
1     Afghanistan  2017  2202.570851
2     Afghanistan  2016  2200.764487
3     Afghanistan  2015  2212.750899
4     Afghanistan  2014  2249.253738
  Country or Area  Year             Value
0     Afghanistan  2014              6.39
1     Afghanistan  2013               5.9
2     Afghanistan  2012  5.45454545454545
3     Afghanistan  2011                 5
4     Afghanistan  2010                 4


11. Change the columns for the GDP Per Capita DataFrame to ‘Country’, ‘Year’, and ‘GDP_Per_Capita’.

In [8]:
gdp_df.columns = ['Country', 'Year', 'GDP_Per_Capita']
print(gdp_df)

            Country                                      Year  GDP_Per_Capita
0       Afghanistan                                      2018     2190.240321
1       Afghanistan                                      2017     2202.570851
2       Afghanistan                                      2016     2200.764487
3       Afghanistan                                      2015     2212.750899
4       Afghanistan                                      2014     2249.253738
...             ...                                       ...             ...
6603       Zimbabwe                                      1992     3054.889178
6604       Zimbabwe                                      1991     3426.598094
6605       Zimbabwe                                      1990     3324.348171
6606  footnoteSeqID                                  Footnote             NaN
6607              2  Excludes South Sudan after July 9, 2011.             NaN

[6608 rows x 3 columns]


12. Change the columns for the Internet Users DataFrame to ‘Country’, ‘Year’, and ‘Internet_Users_Pct’.

In [9]:
internet_df.columns = ['Country', 'Year', 'Internet_Users_Pct']
print(internet_df)

          Country                                               Year  \
0     Afghanistan                                               2014   
1     Afghanistan                                               2013   
2     Afghanistan                                               2012   
3     Afghanistan                                               2011   
4     Afghanistan                                               2010   
...           ...                                                ...   
4671          176                                   NTIA/CPS survey.   
4672          177                     Pew Internet Project, Internet   
4673          178  Estimated based on Survey's results. Populatio...   
4674          179                     Preliminary. Country estimate.   
4675          180  The methodology used to estimated the figure f...   

                        Internet_Users_Pct  
0                                     6.39  
1                                      5.9  


13. Merge the two DataFrames to one. Merge all rows from each of the two DataFrames. Call the new DataFrame gdp_and_internet_use.

In [11]:
gdp_and_internet_use = gdp_df.merge(internet_df, how='outer', on=['Country', 'Year'])

14. Look at the first five rows of your new DataFrame to confirm it merged correctly.

In [12]:
print(gdp_and_internet_use.head(10))
print(gdp_df.head(10))
print(internet_df.head(10))

       Country  Year  GDP_Per_Capita Internet_Users_Pct
0  Afghanistan  2018     2190.240321                NaN
1  Afghanistan  2017     2202.570851                NaN
2  Afghanistan  2016     2200.764487                NaN
3  Afghanistan  2015     2212.750899                NaN
4  Afghanistan  2014     2249.253738               6.39
5  Afghanistan  2013     2264.317575                5.9
6  Afghanistan  2012     2220.481526   5.45454545454545
7  Afghanistan  2011     2037.608607                  5
8  Afghanistan  2010     2093.743565                  4
9  Afghanistan  2009     1881.777626               3.55
       Country  Year  GDP_Per_Capita
0  Afghanistan  2018     2190.240321
1  Afghanistan  2017     2202.570851
2  Afghanistan  2016     2200.764487
3  Afghanistan  2015     2212.750899
4  Afghanistan  2014     2249.253738
5  Afghanistan  2013     2264.317575
6  Afghanistan  2012     2220.481526
7  Afghanistan  2011     2037.608607
8  Afghanistan  2010     2093.743565
9  Afghanistan

15. Look at the last five rows to make sure the data is clean and as expected.

In [14]:
print(gdp_and_internet_use.tail(10))
print(gdp_df.tail(10))
print(internet_df.tail(10))

     Country                                               Year  \
7836     171                        Internet Dial-up customers.   
7837     172  Population age 16+ using the Internet in the l...   
7838     173  Population age 16+ using internet in the last ...   
7839     174  U.S. Census Bureau, Table 2. Reported Internet...   
7840     175             Includes individuals 3 years and older   
7841     176                                   NTIA/CPS survey.   
7842     177                     Pew Internet Project, Internet   
7843     178  Estimated based on Survey's results. Populatio...   
7844     179                     Preliminary. Country estimate.   
7845     180  The methodology used to estimated the figure f...   

      GDP_Per_Capita                    Internet_Users_Pct  
7836             NaN                                   NaN  
7837             NaN                                   NaN  
7838             NaN                                   NaN  
7839             N

16. Subset the combined DataFrame to keep only the data for 2004, 2009, and 2014. Check that this happened correctly.

In [28]:
gdp_and_internet_use[(gdp_and_internet_use.Year == "2004") | (gdp_and_internet_use.Year == "2009") | (gdp_and_internet_use.Year == "2014")]

Country  Year  GDP_Per_Capita Internet_Users_Pct
4               Afghanistan  2014     2249.253738               6.39
9               Afghanistan  2009     1881.777626               3.55
14              Afghanistan  2004     1284.127338  0.105809030021958
21                  Albania  2014    11828.232264               60.1
26                  Albania  2009    10528.778102               41.2
...                     ...   ...             ...                ...
7629  Virgin Islands (U.S.)  2009             NaN   27.3965096846662
7634  Virgin Islands (U.S.)  2004             NaN   27.3770087880198
7646                  Yemen  2014             NaN              22.55
7651                  Yemen  2009             NaN               9.96
7656                  Yemen  2004             NaN  0.881222988679516

[830 rows x 4 columns]

17. Create three new DataFrames, one for 2004, one for 2009, and one for 2014. Give them meaningful names that aren't too long.

In [31]:
df_2004 = gdp_and_internet_use[gdp_and_internet_use.Year == "2004"]
df_2009 = gdp_and_internet_use[gdp_and_internet_use.Year == "2009"]
df_2014 = gdp_and_internet_use[gdp_and_internet_use.Year == "2014"]
print(df_2004)
print(df_2009)
print(df_2014)

                    Country  Year  GDP_Per_Capita Internet_Users_Pct
14              Afghanistan  2004     1284.127338  0.105809030021958
31                  Albania  2004     7738.542836   2.42038779776014
60                  Algeria  2004     9898.310490   4.63447508776537
89                   Angola  2004     5568.959139  0.464814617985909
118     Antigua and Barbuda  2004    18742.463741   24.2665437161785
...                     ...   ...             ...                ...
7532     T.F.Y.R. Macedonia  2004             NaN              24.44
7590              Venezuela  2004             NaN   8.40446959241215
7614               Viet Nam  2004             NaN   7.64240852842363
7634  Virgin Islands (U.S.)  2004             NaN   27.3770087880198
7656                  Yemen  2004             NaN  0.881222988679516

[276 rows x 4 columns]
                    Country  Year  GDP_Per_Capita Internet_Users_Pct
9               Afghanistan  2009     1881.777626               3.55
26        

18. Which country had the highest percentage of internet users in 2014? What was the percentage? (Try typing the first 3 letters of your DataFrame name and hitting the tab key for auto-complete options).

In [38]:
df_2014.sort_values("Internet_Users_Pct", ascending = False)

Country  Year  GDP_Per_Capita Internet_Users_Pct
2507              Iceland  2014    49911.069825              98.16
6651              Bermuda  2014             NaN               96.8
4461               Norway  2014    61308.763730               96.3
1374              Denmark  2014    52172.606425              95.99
6611              Andorra  2014             NaN               95.9
...                   ...   ...             ...                ...
6179               Tuvalu  2014     3455.741510                NaN
6353  Upper middle income  2014    14271.692393                NaN
6469              Vietnam  2014     6098.489609                NaN
6498   West Bank and Gaza  2014     5499.415939                NaN
6523                World  2014    15181.246955                NaN

[277 rows x 4 columns]

ANS: Iceland with 98.16%

19. Which country had the lowest percentage of internet users in 2014? What was the percentage?

In [39]:
df_2014.sort_values("Internet_Users_Pct")

Country  Year  GDP_Per_Capita Internet_Users_Pct
6840  Dem. People's Rep. of Korea  2014             NaN                  0
6898                      Eritrea  2014             NaN               0.99
5978                  Timor-Leste  2014     3264.739189               1.14
770                       Burundi  2014      886.235777               1.38
7435                      Somalia  2014             NaN               1.63
...                           ...   ...             ...                ...
6179                       Tuvalu  2014     3455.741510                NaN
6353          Upper middle income  2014    14271.692393                NaN
6469                      Vietnam  2014     6098.489609                NaN
6498           West Bank and Gaza  2014     5499.415939                NaN
6523                        World  2014    15181.246955                NaN

[277 rows x 4 columns]

Democratic People's Republic of Korea with 0%

20. Repeat for 2004 and 2009.

In [43]:
df_2004.sort_values("Internet_Users_Pct", ascending = False)

Country  Year  GDP_Per_Capita Internet_Users_Pct
494                Belize  2004     7327.474691                9.8
1138             Colombia  2004     9814.268838    9.1186903029666
3337              Lebanon  2004    14283.959712                  9
5785               Sweden  2004    45029.814344              83.89
2517              Iceland  2004    45380.965464              83.88
...                   ...   ...             ...                ...
6189               Tuvalu  2004     3341.738017                NaN
6363  Upper middle income  2004     8561.878030                NaN
6479              Vietnam  2004     3689.732423                NaN
6508   West Bank and Gaza  2004     4113.246936                NaN
6533                World  2004    12070.457345                NaN

[276 rows x 4 columns]